userspace/libsinsp/logger.cpp

```diff
@@ -206,20 +206,34 @@ void sinsp_logger::log(std::string msg, const severity sev)
 	}
 }
 
-const char* sinsp_logger::format(const severity sev, const char* const fmt, ...)
+void sinsp_logger::format(const severity sev, const char* const fmt, ...)
 {
+	if(sev > m_sev)
+	{
+		return;
+	}
+
 	va_list ap;
 
 	va_start(ap, fmt);
 	vsnprintf(s_tbuf, sizeof s_tbuf, fmt, ap);
 	va_end(ap);
 
 	log(s_tbuf, sev);
+}
 
-	return s_tbuf;
+void sinsp_logger::format(const char* const fmt, ...)
+{
+	va_list ap;
+
+	va_start(ap, fmt);
+	vsnprintf(s_tbuf, sizeof s_tbuf, fmt, ap);
+	va_end(ap);
+
+	log(s_tbuf, SEV_INFO);
 }
 
-const char* sinsp_logger::format(const char* const fmt, ...)
+const char* sinsp_logger::format_and_return(const severity sev, const char* const fmt, ...)
```

Comment:
If there's no client for this, then I'd just git rid of it.

Reply:
test uses it to validate something useful. may be other ways to do the validation, but don't need to rewrite the world for this.